core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -4107,6 +4113,97 @@ public static List arrayExcept(List list1, List list2) {
     return new ArrayList<>(result);
   }
 
+  /** Support the ARRAY_INSERT(array, pos, val) function. */
+  public static @Nullable List arrayInsert(List baselist, Object pos, Object val) {
```

Comment:
@NobiGo e.g. array_insert(array(1,2,3), 1, 2) calcite will parse and search `arrayInsert[interface java.util.List, int, int]` signature, but it can't match `arrayInsert(List baselist, Integer pos, Object val)`. Java method reflection limits boxing and primitive types can't be matched.

we can other array function. i.e. array_position has same behavior.

Reply:
another case, if second pos is a null value. e.g. array_insert(array(1,2,3), cast(null as Integer), 2), then searched method is `arrayInsert[interface java.util.List, Integer, int]`.  We want to use `Integer` to cover `int and Integer`, but it can't be work.